daemon/daemon_test.go

```diff
@@ -98,18 +97,58 @@ func testProxyRequest(t *testing.T, path string, expectedStatus int, expectedBod
 	}
 }
 
+func getHealthCheckURLFromPort(port string) string {
+	if len(port) > 0 {
+		return "http://localhost:" + port + "/health/check"
+	}
+	return "http://localhost/health/check"
+}
+
 func TestHealthCheck(t *testing.T) {
-	err := setUpDaemonWithLocalServer()
+	localProxyListen := ":6634"
+	healthCheckPort := "60002"
+
+	err := setUpDaemonWithLocalServer("8000", localProxyListen, healthCheckPort)
 	if err != nil {
 		t.Fatalf("Test daemon setup failed: %s", err.Error())
 	}
 
+	localProxyURL := "http://localhost" + localProxyListen
+
 	// Test a route that should be proxied to 404.
-	testProxyRequest(t, "/helloworld", http.StatusNotFound, "404")
+	testProxyRequest(t, localProxyURL+"/helloworld", http.StatusNotFound, "404")
 
 	// Test a route that should be proxied to a valid response.
-	testProxyRequest(t, "/healthyroute", http.StatusOK, "healthy")
+	testProxyRequest(t, localProxyURL+"/healthyroute", http.StatusOK, "healthy")
 
 	// Test the health check.
-	testProxyRequest(t, "/sphinx/health/check", http.StatusOK, "")
+	testProxyRequest(t, getHealthCheckURLFromPort(healthCheckPort), http.StatusOK, "")
+}
+
+func TestDaemonWithNoHealthCheck(t *testing.T) {
+	localProxyListen := ":6635"
+	healthCheckPort := ""
+
+	err := setUpDaemonWithLocalServer("8001", localProxyListen, healthCheckPort)
+	if err != nil {
+		t.Fatalf("Test daemon setup failed: %s", err.Error())
+	}
+
+	// Because so many servers are starting,  sleep  for a second to make sure
+	// they start.
+	time.Sleep(time.Second)
+
+	localProxyURL := "http://localhost" + localProxyListen
+
+	// Test a route that should be proxied to 404.
+	testProxyRequest(t, localProxyURL+"/helloworld", http.StatusNotFound, "404")
+
+	// Test a route that should be proxied to a valid response.
+	testProxyRequest(t, localProxyURL+"/healthyroute", http.StatusOK, "healthy")
```

Comment:
Should I pull lines 128 - 147 out into a common function?  It's the same as in the TestHealthCheck function.